### PR TITLE
Make schema in signals optional

### PR DESCRIPTION
**Pull request by dsmilkov**
Signal authors can now omit providing schema. In that case the schema will be auto-inferred from the emitted items.

### CODE REVIEW

**Review by nsthorat — APPROVED** [review]
nice work

File: lilac/data/dataset_duckdb.py [hunk 1]
```diff
@@ -1919,6 +1929,7 @@ def select_rows_schema(
       if col.signal_udf:
         udfs.append(SelectRowsSchemaUDF(path=dest_path, alias=col.alias))
         field = col.signal_udf.fields()
+        assert field, f'Expected signal {col.signal_udf.name} to have a schema defined.'
```
[the remark this answers is not in the document]
Comment (dsmilkov) [reply lilac/data/dataset_duckdb.py:1932]: done

File: lilac/data/dataset_utils.py [hunk 2]
```diff
@@ -111,14 +111,19 @@ def schema_contains_path(schema: Schema, path: PathTuple) -> bool:
   return True
 
 
-def create_signal_schema(signal: Signal, source_path: PathTuple, current_schema: Schema) -> Schema:
+def create_signal_schema(
+  signal: Signal, source_path: PathTuple, current_schema: Schema
+) -> Optional[Schema]:
   """Create a schema describing the enriched fields added an enrichment."""
```
[the remark this answers is not in the document]
Comment (dsmilkov) [reply lilac/data/dataset_utils.py:117]: done

File: lilac/data/dataset_compute_signal_test.py [hunk 3]
```diff
@@ -718,3 +718,21 @@ def test_concept_signal_with_select_groups(make_test_data: TestDataMaker) -> Non
     f'text.{concept_key}.*.score', sort_by=GroupsSortBy.COUNT, sort_order=SortOrder.ASC
   )
   assert result.counts == [('In concept', 1), ('Not in concept', 2)]
+
+
+def test_optional_schema(make_test_data: TestDataMaker) -> None:
```
[the remark this answers is not in the document]
Comment (dsmilkov) [reply lilac/data/dataset_compute_signal_test.py:723]: done

**Pull request merged**